src/k-provider/ott/provider.js

```diff
@@ -129,28 +116,109 @@ export default class OTTProvider extends BaseProvider<OTTProviderMediaInfoObject
             });
           }
           const mediaEntry = OTTProviderParser.getMediaEntry(response, requestData);
-          const mediaSources = mediaEntry.sources.toJSON();
-          mediaConfig.sources.hls = mediaSources.hls;
-          mediaConfig.sources.dash = mediaSources.dash;
-          mediaConfig.sources.progressive = mediaSources.progressive;
-          mediaConfig.sources.id = mediaEntry.id;
-          mediaConfig.sources.duration = mediaEntry.duration;
-          mediaConfig.sources.type = mediaEntry.type;
-          mediaConfig.sources.dvr = !!mediaEntry.dvrStatus;
-          mediaConfig.sources.poster = mediaEntry.poster;
-          if (
-            mediaEntry.metadata &&
-            mediaEntry.metadata.metas &&
-            typeof mediaEntry.metadata.metas.tags === 'string' &&
-            mediaEntry.metadata.metas.tags.indexOf('360') > -1
-          ) {
-            mediaConfig.sources.vr = {};
-          }
-          Object.assign(mediaConfig.sources.metadata, mediaEntry.metadata);
+          Object.assign(mediaConfig.sources, this._getSourcesObject(mediaEntry));
         }
       }
     }
     this._logger.debug('Data parsing finished', mediaConfig);
     return mediaConfig;
   }
+
+  /**
+   * Gets playlist config from entry list.
+   * @param {ProviderEntryListObject} entryListInfo - ott entry list info
+   * @returns {Promise<ProviderPlaylistObject>} - The provider playlist config
+   */
+  getEntryListConfig(entryListInfo: ProviderEntryListObject): Promise<ProviderPlaylistObject> {
+    if (entryListInfo.ks) {
+      this.ks = entryListInfo.ks;
+      this._isAnonymous = false;
+    }
+    this._dataLoader = new OTTDataLoaderManager(this.partnerId, this.ks, this._networkRetryConfig);
+    return new Promise((resolve, reject) => {
+      const entries = entryListInfo.entries;
+      if (entries && entries.length) {
+        let ks: string = this.ks;
+        if (!ks) {
+          ks = '{1:result:ks}';
+          this._dataLoader.add(OTTSessionLoader, {partnerId: this.partnerId});
+        }
+        this._dataLoader.add(OTTAssetListLoader, {entries, ks});
+        this._dataLoader.fetchData().then(
+          response => {
+            resolve(this._parseEntryListDataFromResponse(response));
+          },
+          err => {
+            reject(err);
```

Comment:
make sure error from API doesn't leave us with unknown error state in player, i.e. endless spinner

Reply:
error works as expected